tests/test_filter.py

```diff
@@ -147,3 +234,34 @@ def test_filter(grid_type_and_input_ds, filter_args):
             filter = Filter(
                 grid_type=grid_type, grid_vars=grid_vars_missing, **filter_args
             )
+
+
+#################### Visosity-based filter tests ########################################
+@pytest.mark.parametrize(
+    "filter_args",
+    [dict(filter_scale=1.0, dx_min=1.0, n_steps=10, filter_shape=FilterShape.TAPER)],
+)
+def test_viscosity_filter(vector_grid_type_and_input_ds, filter_args):
+    """Test all viscosity-based filters: filters that use a vector Laplacian."""
+    grid_type, da_u, da_v, grid_vars, geolat_u = vector_grid_type_and_input_ds
+
+    filter = Filter(grid_type=grid_type, grid_vars=grid_vars, **filter_args)
+    filtered_u, filtered_v = filter.apply_to_vector(da_u, da_v, dims=["y", "x"])
+
+    # check conservation under solid body rotation: u = cos(lat), v=0;
+    # this test fails due to numerical instabilities
+    # data_u = np.cos(geolat_u/360*2*np.pi)
+    # data_v = np.zeros_like(data_u)
+    # da_u = xr.DataArray(data_u, dims=["y", "x"])
+    # da_v = xr.DataArray(data_v, dims=["y", "x"])
+    # filtered_u, filtered_v = filter.apply_to_vector(da_u, da_v, dims=["y", "x"])
+    # xr.testing.assert_allclose(filtered_u, 0.0, atol=1e-12)
+    # xr.testing.assert_allclose(filtered_v, 0.0, atol=1e-12)
```

Comment:
Why commented-out code? Probably best to avoid this.

Reply:
I fixed the test and uncommented it. 

(I had wrongly assumed that numerical instability had caused the test to fail, but it was actually a small bug in the test - which is now fixed.)